Build and push container images and artifacts on tagged releases

mobileoverlord opened the pull request:
Performs the following when a tag is pushed
* Create a new Release if one doesn't already exist
* Pull release notes from CHANGELOG.md for the tag listed as `## $TAG` IE `## v2.5.4`
* Build arch specific containers for `arm64` and `amd64` `peridiod` release bundles and upload to the release page
* Build arch specific containers for `arm64` and `amd64` and push to docker hub
* Combine docker arch specific images into `peridio/peridiod:$TAG` and `peridio/peridiod/latest` to make a multiarch image
* Push

danielspofford reviewed (changes requested):
Actually I think we can probably cut the size of this workflow almost in half if you use a build matrix. See: https://github.com/peridio/peridio-cli/blob/main/.github/workflows/release.yml#L15. You can specify more than one key for one entry of the build matrix.

mobileoverlord commented:
@danielspofford Updated to leverage martix builds. Also moved the release container files to a `release` directory. Changed the defaults to remove the creation of a peridio/peridio user by default and removed the default uboot environment. Tested using the PeridioEVK.
